src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/MsaPassthroughHandler.cs

```diff
@@ -80,16 +80,54 @@ public async Task<string> TryFetchTransferTokenAsync(AuthenticationRequestParame
                 return null;
             }
 
-            _ = _msaPlugin.ParseSuccessfullWamResponse(transferResponse.ResponseData[0], out var properties);
+            return ExtractTransferTokenAsync(authenticationRequestParameters.AppConfig.ClientId, transferResponse);
+        }
 
-            properties.TryGetValue("code", out string code);
+        public async Task<string> TryFetchTransferSilentDefaultAccountAsync(AuthenticationRequestParameters authenticationRequestParameters, WebAccountProvider accountProvider)
+        {
+            // Apps can have MSA-PT enabled and can configured to allow MSA users
+            // However, some older apps have 2 incarnations, one in AAD tenant and one in MSA tenant
+            // For this second case, we can't fetch the transfer token from the client_ID in AAD and this will fail
+            _logger.Verbose("WAM MSA-PT - fetching transfer token for interactive flow");
 
+            var webTokenRequestMsa = await _msaPlugin.CreateWebTokenRequestAsync(
+                accountProvider,
+                authenticationRequestParameters,
+                isForceLoginPrompt: false,
+                isInteractive: false,
+                isAccountInWam: true,
+                TransferTokenScopes)
+               .ConfigureAwait(false);
 
-            // Important: cannot use this WebAccount with the AAD provider
-            WebAccount msaPtWebAccount = transferResponse.ResponseData[0].WebAccount;
-            _logger.InfoPii($"Obtained a transfer token for {msaPtWebAccount.UserName} ?  {code != null}", $"Obtained a transfer token? {code != null}");
+            WamAdapters.AddMsalParamsToRequest(authenticationRequestParameters, webTokenRequestMsa, _logger);
 
-            return code;
+            var transferResponse = await _wamProxy.GetTokenSilentlyForDefaultAccountAsync(webTokenRequestMsa)
+                .ConfigureAwait(true);
+
+            if (!transferResponse.ResponseStatus.IsSuccessStatus())
+            {
+                try
+                {
+                    _ = WamAdapters.CreateMsalResponseFromWamResponse(
+                        transferResponse,
+                        _msaPlugin,
+                        authenticationRequestParameters.AppConfig.ClientId,
+                        _logger,
+                        isInteractive: true);
```

Comment:
Does the value of this parameter matter? This `Extract` method is called from ATS methods.